src/components/tl/ucp/allreduce/allreduce_knomial.c

```diff
@@ -106,7 +106,7 @@ ucc_status_t ucc_tl_ucp_allreduce_knomial_progress(ucc_coll_task_t *coll_task)
             } else {
                 send_buf = rbuf;
             }
-            ucc_tl_ucp_send_nb(send_buf, data_size, mem_type, peer, team, task);
+            CHECK_STATUS(ucc_tl_ucp_send_nb(send_buf, data_size, mem_type, peer, team, task), task);
```

Comment:
need to free memory allocated through memory component in case of error

Reply:
Added "goto label" and run through clang-format